Documentation/RepoCompose.md

```diff
@@ -0,0 +1,287 @@
+    "floating": { 
+        // Build artifacts
+    },
+    "static": { 
+        // Static artifacts
+    },
+    "toolset": { 
+        // Toolset artifacts
+    }
+}
+```
+
+The data in the output is further grouped by operating system.  Builds of the same repo on different operating system can reasonably consume a different set of resources.  The output of `consumes` reflect this and allows per operating system dependencies:
+
+``` json
+{
+    "os-windows": {
+        "dependencies": { } 
+    },
+    "os-linux": {
+        "dependencies": { }
+    }
+}
+```
+
+In the case the `consumes` output doesn't want to take advantage of OS specific dependencies it can specify `"os-all"` as a catch all. 
+
+In addition to artifacts the consume feed can also optionally list any machine prerequitsites needed to build or test the repo:
+
+``` json
+"prereq": { 
+    "Microsoft Visual Studio": "2015",
+    "CMake" : "1.0",
+}
+```
+
+A full sample output for `consumes` is available in the Samples section.
+
+### produces
+
+The `produces` command returns json output which describes the artifacts produced by the the repo.  This includes NuGet packages and file artifacts.  
+
+The output format for artifacts is special for `produces` because it lacks any hard location information.  For example: 
+
+- NuGet artifacts lack feeds
+- File artifacts lack a `"kind"` and supporting values
+
+This is because the `produces` command represents "what" a repo produces, not "where" the repo produces it.  The "where" portion is controled by the `publish` command.  External components will take the output of `produces`, add the location information in and feed it back to `publish`.  
+
+Like `consumes` the `produces` output is also grouped by the operating system:
+
+``` json
+{
+    "os-windows": {
+        "nuget": { },
+        "file": { }
+    },
+    "os-linux": {
+        "nuget": { },
+        "file": { }
+    }
+}
+```
+
+A ful sample output for `produces` is available in the Samples section.
+
+### change
+
+The `change` command is used to alter the floating build dependencies section.  It can establish new versions of NuGet packages, new locations to find file artifacts, different NuGet feeds, etc ...  
+
+This is the command which allow us to use the build output of one repo as the input of a dependent repo.  The first repo can build using a new output version (say beta5) and dependent repos can be changed to accept this new version.  
+
+This command operates by providing json whose format is a subset of the output of `consumes`.  In particular it will provide the `"dependencies.floating"` section.  
+
+``` json
+"dependencies" {
+    "floating": { 
+        "nuget": {
+            "packages" {
+                "MicroBuild.Core": "0.2.0",
+                "Microsoft.NETCore.Platforms": "1.0.1"
+            }
+        }
+    }
+}
+```
+
+A tool responsible for composing repos would ideally:
+
+1. Execute `run.cmd consumes` and capture the output.
```

Comment:
Could you please add a graphic of how the commands interact with each other? I think that would help to understand how the flow and dependencies of the commands are.